content/en/llm_observability/setup/sdk/nodejs.md

```diff
@@ -0,0 +1,713 @@
+function getRelevantDocs (question) {
+  const contextDocuments = ... // user application logic
+  llmobs.annotate({
+    inputData: question,
+    outputData: contextDocuments.map(doc => ({
+      id: doc.id,
+      score: doc.score,
+      text: doc.text,
+      name: doc.name
+    }))
+  })
+  return
+}
+getRelevantDocs = llmobs.wrap('retrieval', getRelevantDocs)
+{{< /code-block >}}
+
+### Conditions for finishing a span for a wrapped function
+
+`llmobs.wrap` extends the underlying behavior of `tracer.wrap`. The underlying span created when the function is called is finished under the following conditions:
+
+1. The function returns a Promise, in which case the span will finish when the promise is resolved or rejected.
+2. The function takes a callback as its last parameter, in which case the span will finish when that callback is called.
+3. The function doesn't accept a callback and doesn't return a Promise, in which case the span will finish at the end of the function execution.
+
+#### Example
+  
+{{< code-block lang="javascript" >}}
+const express = require('express')
+const app = express()
+
+function myAgentMiddleware (req, res, next) {
+  const err = ... // user application logic
+  // the span for this function is finished when `next` is called
+  next(err)
+}
+myAgentMiddleware = llmobs.wrap('agent', myAgentMiddleware)
+
+app.use(myAgentMiddleware)
+
+{{< /code-block >}}
+
+If the application does not use the callback function, it is recommended to use an inline traced block instead. See [Tracing spans using inline methods](#tracing-spans-using-inline-methods) for more information.
+
+{{< code-block lang="javascript" >}}
+const express = require('express')
+const app = express()
+
+function myAgentMiddleware (req, res) {
+  // the `next` callback is not being used here
+  return llmobs.trace('agent', { name: 'myAgentMiddleware' }, () => {
+    return res.status(200).send('Hello World!')
+  })
+}
+myAgentMiddleware = llmobs.wrap('agent', myAgentMiddleware)
+
+app.use(myAgentMiddleware)
+
+{{< /code-block >}}
+
+## Tracking user sessions
+
+Session tracking allows you to associate multiple interactions with a given user. When starting a root span for a new trace or span in a new process, specify the `session_id` argument with the string ID of the underlying user session:
+
+{{< code-block lang="javascript" >}}
+function processMessage() {
+    ... # user application logic
+    return
+}
+processMessage = llmobs.wrap('workflow', { sessionId: "<SESSION_ID>" }, processMessage)
+{{< /code-block >}}
+
+## Annotating a span
+
+The SDK provides the method `llmobs.annotate()` to annotate spans with inputs, outputs, and metadata. 
+
+### Arguments
+
+The `LLMObs.annotate()` method accepts the following arguments:
+
+`span` 
+: optional - _Span_ - **default**: the current active span
+<br />The span to annotate. If `span` is not provided (as when using function wrappers), the SDK annotates the current active span.
```

Comment:
Question (not doc related): does the Node SDK annotate the current active LLMOBS-type span? Or would this conflict with APM/LLMOBS type spans?

Reply:
yeah it would conflict if the current span was not started by the SDK/LLMObs integration. in that case, it logs and returns early.

Reply:
Cool, we'll talk more about context management next week to ensure we'll always find the current active LLMObs span instead of this current behavior